valhalla/thor/costmatrix.h

```diff
@@ -95,11 +97,17 @@ struct BestCandidate {
  */
 class CostMatrix {
  public:
+
   /**
-   * Constructor with cost threshold.
-   * @param initial_cost_threshold  Cost threshold for termination.
+   * Constructor with cost threshold. Use custom cost threshold is set to true.
+   * @param auto_cost_threshold        Cost threshold for termination using auto mode.
+   * @param bicycle_cost_threshold     Cost threshold for termination using bicycle mode.
+   * @param pedestrian_cost_threshold  Cost threshold for termination using pedestrian mode.
    */
-  CostMatrix(float initial_cost_threshold = kCostThresholdDefault);
+  CostMatrix(
+      float auto_cost_threshold = kDefaultCostThresholdAuto,
+      float bicycle_cost_threshold = kDefaultCostThresholdBicycle,
+      float pedestrian_cost_threshold = kDefaultCostThresholdPedestrian);
```

Comment:
Why are you passing in 3 different thresholds? The costing model tells which one to use. There is no support for multimodal matrices so only one mode at a time is supported. Also, shouldn't the threshold be based on the max service limit?